File: PO/Projekt02__konsolowe__Analizator_tekstu_z_pliku/python/main.py
import sys
import os
import string


def main():
    try:
        VOWELS = ["a", "e", "i", "y", "o", "u", "A", "E", "I", "O", "Y", "U"]

        fileName: str

        if len(sys.argv) >= 2:
            fileName = sys.argv[1]
        else:
            fileName = input("Podaj sciezke do pliku: ")

        if len(fileName) == 0:
            raise Exception("File name was not provide!")

        charsCount: int = 0
        linesCount: int = 0
        vowelsCount: int = 0
        cononantsCount: int = 0
        wordsCount = 0

        with open(fileName, "r") as f:
            for line in f.readlines():
                linesCount += 1
                words = [word for word in line.split(" ") if word not in ["\n", ""]]

                wordsCount += len(words)

                for char in line:
                    if char != "\n":
                        charsCount += 1
                    if char in VOWELS:
                        vowelsCount += 1
                    elif char not in string.whitespace:
                        cononantsCount += 1

        print(f"size of file: {os.stat(fileName).st_size} bytes")
        print(f"count of words: {wordsCount}")
        print(f"count of lines: {linesCount}")
        print(f"count of characters (with spaces): {charsCount}")

        print("------------------------")
        print("Without whitespaces!")
        print("------------------------")

        print(f"count of vowels: {vowelsCount}")
        print(f"count of consonants: {cononantsCount}")

    except FileExistsError:
        print("[ERROR]: File not exist!")
    except FileNotFoundError:
        print("[ERROR]: File not found!")
    except Exception as ex:
        print(f"[ERROR]: {ex}")


if __name__ == "__main__":
    main()

File: PO/Projekt02__konsolowe__Analizator_tekstu_z_pliku/cpp/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <sstream>
#include <algorithm>
#include <filesystem>

using namespace std;
namespace fs = std::filesystem;

int main(int argc, char *argv[])
{
    std::vector<char> vowels = {'a', 'e', 'i', 'y', 'o', 'u', 'A', 'E', 'I', 'O', 'Y', 'U'};
    try
    {
        string fileName;
        if (argc >= 2)
            fileName = argv[1];
        else
        {
            cout << "Podaj sciezke do pliku: ";
            getline(std::cin, fileName);
        }

        {
            if (fileName.length() == 0 || fileName.empty())
                throw 400;

            if (!fs::exists(fileName))
                throw 404;

            string line;
            ifstream File(fileName);

            int charsCount = 0;
            int linesCount = 0;
            int vowelsCount = 0;
            int consonantsCount = 0;
            int wordsCount = 0;

            while (getline(File, line))
            {
                ++linesCount;

                stringstream ss(line);
                string word;

                while (ss >> word)
                    ++wordsCount;

                for (char c : line)
                {
                    ++charsCount;

                    if (std::find(vowels.begin(), vowels.end(), c) != vowels.end())
                        vowelsCount++;
                    else if (c != ' ')
                        consonantsCount++;
                }
            }

            cout << "size of file: " << fs::file_size(fileName) << " bytes" << endl;

            cout << "count of words: " << wordsCount << endl;
            cout << "count of lines: " << linesCount << endl;
            cout << "count of characters (including spacebars): " << charsCount << endl;

            cout << "------------------------" << endl;
            cout << "Without spacebars!" << endl;
            cout << "------------------------" << endl;

            cout << "count of vowels: " << vowelsCount << endl;
            cout << "count of consonants: " << consonantsCount << endl;

            File.close();
        }

        return 0;
    }
    catch (const int &e)
    {
        if (e == 400)
            cerr << "File name was not provide" << endl;
        if (e == 404)
            cerr << "File Not Exist!" << endl;
    }
}